File: frontend/src/types/aiRiskSummary.ts
/** AI risk-summary types — field-level mirror of backend AIRiskSummaryRead.
 *
 * There is no risk-score field: EventRisk.score stays the single official
 * score; the AI summary only explains, compresses and prioritises.
 */

export type AnalystPriority = 'low' | 'medium' | 'high' | 'critical'

export interface AIRiskSummary {
  id: string
  alert_group_id: string
  provider: string
  model: string
  summary: string
  key_findings: string[]
  risk_drivers: string[]
  analyst_priority: AnalystPriority
  confidence: number
  created_at: string
  updated_at: string
}

File: backend/app/schemas/ai_risk_summary.py
"""Pydantic schemas of the AI risk-summary API."""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AIRiskSummaryRead(BaseModel):
    """One AI risk summary as returned by the API.

Mirrors the ai_risk_summaries row field-for-field, so the response
carries the structured-output protocol (summary / key_findings /
risk_drivers / analyst_priority / confidence) that every provider must
produce. The payload holds no risk score: EventRisk.score remains the
authoritative one.
"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    alert_group_id: uuid.UUID
    provider: str
    model: str
    summary: str
    key_findings: list[str]
    risk_drivers: list[str]
    analyst_priority: str
    confidence: float
    created_at: datetime
    updated_at: datetime
